api/test_cache.py

```diff
@@ -0,0 +1,127 @@
+from django.test import override_settings
+from django.contrib.auth.models import User, AnonymousUser
+from django.conf import settings
+from fakeredis import FakeConnection
+
+from main.settings import *
+from main.test_case import APITestCase
+from api import models
+
+FAKE_REDIS_CACHE = {
+    'default': {
+        'BACKEND': 'django_redis.cache.RedisCache',
+        'LOCATION': env('CACHE_REDIS_URL'),
+        'OPTIONS': {
+            'CLIENT_CLASS': 'django_redis.client.DefaultClient',
+            'connection_class': FakeConnection
+        }
+    }
+}
```

Comment:
It should be fine to use a Memory cache for testing as we aren't using any Redis-specific features.
```python
    'default': {
        'BACKEND': 'django.core.cache.backends.locmem.LocMemCache',
        'LOCATION': 'unique-snowflake-rest',
    }
```

Or we can use a different database index for testing (max supported is 16 by default)
For eg.
Define env variables for testing for docker-compose and then use them in settings.py and here
```bash
CELERY_REDIS_URL: ${CELERY_REDIS_URL:-redis://redis:6379/0}
CELERY_TEST_REDIS_URL: ${CELERY_TEST_REDIS_URL:-redis://redis:6379/10}
CACHE_REDIS_URL: ${CACHE_REDIS_URL:-redis://redis:6379/1}
CACHE_TEST_REDIS_URL: ${CACHE_TEST_REDIS_URL:-redis://redis:6379/11}
```
And  make TEST_* as optional in settings.py

Reply:
These tests don't actually use Redis. I'm using the fakeredis library, hence the `FakeConnection` connection class. But I can use Redis instead if that's preferred. 

I decided against `LocMemCache` to keep the test environment close to production. We're not using anything Redis-specific now but we might in the future.

Reply:
Then, using Redis directly makes sense here.
Want to avoid adding a new library that simulates a service that we already have running :D

Reply:
Ok. I changed that so the cache tests use Redis and removed the dependency